eq_translations/survey_schema.py

```diff
@@ -1,245 +1,204 @@
 
     def __init__(self, schema_data=None):
         self.schema = schema_data
-        self.load_placeholders()
 
     def load(self, schema_path):
-        with open(schema_path, encoding="utf8") as schema_file:
-            self.schema = json.load(schema_file)
-            self.load_placeholders()
-
-    def load_placeholders(self):
-        if self.schema:
-            (
-                self.context_placeholder_pointers,
-                self.no_context_placeholder_pointers,
-            ) = self.get_placeholder_pointers()
+        with open(schema_path, encoding="utf8") as schema_file:  # pragma: no cover
+            self.schema = json.load(schema_file)  # pragma: no cover
 
     def save(self, schema_path):
-        with open(schema_path, "w", encoding="utf8") as schema_file:
-            return json.dump(self.schema, schema_file, indent=4)
-
-    @property
-    def pointers(self):
-        return self.no_context_pointers + self.context_pointers
-
-    @property
-    def no_context_pointers(self):
-        return (
-            self.get_core_pointers()
-            + self.get_title_pointers()
-            + self.get_message_pointers()
-            + self.get_list_pointers()
-            + self.no_context_placeholder_pointers
-        )
+        with open(schema_path, "w", encoding="utf8") as schema_file:  # pragma: no cover
+            return json.dump(self.schema, schema_file, indent=4)  # pragma: no cover
 
     @property
-    def context_pointers(self):
-        return self.get_answer_pointers() + self.context_placeholder_pointers
-
-    def get_core_pointers(self):
+    def translatable_strings(self):
         """
-        Iterate the schema and return the pointers found
-        :return:
+        :yield: A pointer dictionary
+        :return: A generator
         """
-        pointers = []
-        for key in SurveySchema.no_context_keys:
-            key_pointers = find_pointers_to(self.schema, key)
-            pointers.extend(key_pointers)
-        return pointers
 
-    def get_placeholder_pointers(self):
-        """
-        Placeholder pointers may have context or may not
-        :return:
-        """
-        found_pointers = find_pointers_to(self.schema, "text")
-        context_placeholder_pointers = []
-        no_context_placeholder_pointers = []
+        for key in self.keys_to_translate:
+            with_context = key in self.keys_with_context
 
-        for pointer in found_pointers:
-            if "/answers/" in pointer:
-                context_placeholder_pointers.append(pointer)
-            else:
-                no_context_placeholder_pointers.append(pointer)
+            for pointer in find_pointers_to(self.schema, key):
+                # The 'list' property is used by both 'contents' and 'when' blocks.
+                if "/when/" in pointer:
+                    continue
 
-        return context_placeholder_pointers, no_context_placeholder_pointers
+                self.total_translatable_strings += 1
```

Comment:
I don't think this will be correct. Both dicts and strings can hold multiple strings for translation. We shouldn't need to count as we go anyway as we can count the number of items in `translatable_strings`.